src/ck3/data/lease_contracts.rs

```diff
@@ -0,0 +1,86 @@
+use crate::block::{Block, BV};
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+use crate::validate::validate_modifiers_with_base;
+use crate::validator::{Validator, ValueValidator};
+
+#[derive(Clone, Debug)]
+pub struct LeaseContract {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Ck3, Item::LeaseContract, LeaseContract::add)
+}
+
+// TODO: verify somewhere that `theocracy_lease` is a defined item.
+
+impl LeaseContract {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::LeaseContract, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for LeaseContract {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+        vd.req_field("hierarchy");
+        vd.field_validated_block("hierarchy", |block, data| {
+            let mut vd = Validator::new(block, data);
+            vd.field_validated_key_block("ruler_valid", |key, block, data| {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                validate_trigger(block, data, &mut sc, Tooltipped::No);
+            });
+            vd.field_validated_key_block("liege_or_vassal_valid", |key, block, data| {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                sc.define_name("target", Scopes::Character, key);
+                validate_trigger(block, data, &mut sc, Tooltipped::No);
+            });
+            vd.field_validated_key_block("barony_valid", |key, block, data| {
+                let mut sc = ScopeContext::new(Scopes::LandedTitle, key);
+                validate_trigger(block, data, &mut sc, Tooltipped::No);
+            });
+            {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                vd.field_target("lessee", &mut sc, Scopes::Character);
+            }
+        });
+
+        vd.field_item("government", Item::GovernmentType);
+        vd.field_list_items("valid_holdings", Item::HoldingType);
+        vd.field_integer("ruler_share_min_opinion_from_lessee");
+        vd.field_choice("hook_strength_max_opinion", &["none", "any", "strong"]);
+
+        for field in &["tax", "levy"] {
```

Comment:
> "Note: Levies can only go to one character on each level. This means levies cannot be assigned in part to the ruler and in part to the lessee because they are considered to be on the same level. So you can use only 0 or 100 when splitting between ruler and lessee."

Reply:
Lol I didn't read past "See tax for syntax"

Reply:
I'm not sure this comment is true. The existing `theocracy_lease` doesn't follow it.

Reply:
I also remember from playing that you get different amounts of troops from the chaplain depending on his opinion of you

Reply:
Yes from the look of it it seems that comment is outdated since clearly they don't follow that as you pointed out.